src/middlewared/middlewared/plugins/pool.py

```diff
@@ -2749,13 +2813,560 @@ def query(self, filters=None, options=None):
         return True
 
 
+class PoolDatasetEncryptionModel(sa.Model):
+    __tablename__ = 'storage_datasetencryption'
```

Comment:
We already have `storage.encrypteddisk`, why can't we name this in same manner? E.g. `storage_encrypteddataset`.